docs/ref/models/meta.txt

```diff
@@ -0,0 +1,277 @@
+         <django.db.models.fields.CharField: username>,
+         <django.db.models.fields.CharField: first_name>,
+         <django.db.models.fields.CharField: last_name>,
+         <django.db.models.fields.EmailField: email>,
+         <django.db.models.fields.BooleanField: is_staff>,
+         <django.db.models.fields.BooleanField: is_active>,
+         <django.db.models.fields.DateTimeField: date_joined>,
+         <django.db.models.fields.related.ManyToManyField: groups>,
+         <django.db.models.fields.related.ManyToManyField: user_permissions>)
+
+        # Also include hidden fields.
+        >>> User._meta.get_fields(include_hidden=True)
+        (<ManyToOneRel: auth.user_groups>,
+         <ManyToOneRel: auth.user_user_permissions>,
+         <ManyToOneRel: admin.logentry>,
+         <django.db.models.fields.AutoField: id>,
+         <django.db.models.fields.CharField: password>,
+         <django.db.models.fields.DateTimeField: last_login>,
+         <django.db.models.fields.BooleanField: is_superuser>,
+         <django.db.models.fields.CharField: username>,
+         <django.db.models.fields.CharField: first_name>,
+         <django.db.models.fields.CharField: last_name>,
+         <django.db.models.fields.EmailField: email>,
+         <django.db.models.fields.BooleanField: is_staff>,
+         <django.db.models.fields.BooleanField: is_active>,
+         <django.db.models.fields.DateTimeField: date_joined>,
+         <django.db.models.fields.related.ManyToManyField: groups>,
+         <django.db.models.fields.related.ManyToManyField: user_permissions>)
+
+.. _migrating-old-meta-api:
+
+Migrating from the old API
+==========================
+
+As part of the formalization of the ``Model._meta`` API (from the
+:class:`django.db.models.options.Options` class), a number of methods and
+properties have been deprecated and will be removed in Django 2.0.
+
+These old APIs can be replicated by either:
+
+* invoking :meth:`Options.get_field()
+  <django.db.models.options.Options.get_field()>`, or;
+
+* invoking :meth:`Options.get_fields()
+  <django.db.models.options.Options.get_fields()>` to retrieve a list of all
+  fields, and then filtering this list using the :ref:`field attributes
+  <model-field-attributes>` that describe (or retrieve, in the case of
+  ``_with_model`` variants) the properties of the desired fields.
+
+Assuming you have a model named ``MyModel``, the following substitutions
+can be made to convert your code to the new API:
+
+* ``MyModel._meta.get_field(name)``::
+
+      f = MyModel._meta.get_field(name)
+
+  then check if:
+
+  - ``f.auto_created == False``, because the new ``get_field()``
+    API will find "reverse" relations), and:
+
+  - ``f.is_relation and f.related_model is None``, because the new
+    ``get_field()`` API will find
+    :class:``~django.contrib.contenttypes.fields.GenericForeignKey``
+    relations;
+
+* ``MyModel._meta.get_field_by_name(name)[0]``::
```

Comment:
It may be useful to address the case where someone doesn't just index the first item of the tuple.